packages/go/bomenc/encodings.go

```diff
@@ -0,0 +1,144 @@
+	return b.encodingType
+}
+
+// Sequence returns the BOM sequence for this encoding.
+// This method fulfills the Encoding interface and provides access to the BOM sequence,
+// which is essential for encoding detection and writing files with proper BOMs.
+func (b bomEncoding) Sequence() []byte {
+	return b.sequence
+}
+
+// HasSequence checks if the given data starts with this encoding's BOM sequence.
+// This method fulfills the Encoding interface and provides a way to check for
+// the presence of this encoding's BOM, which is crucial for encoding detection.
+func (b bomEncoding) HasSequence(data []byte) bool {
+	return b.hasSequenceFunc(data)
+}
+
+// The following functions are used to check for specific encoding BOMs.
+// By defining these as separate functions, we can easily reuse them
+// and potentially extend them if more complex checking is needed in the future.
+// This approach also keeps the bomEncoding struct clean and simple.
+
+func isUTF32BE(buf []byte) bool {
+	return len(buf) >= 4 && buf[0] == 0x00 && buf[1] == 0x00 && buf[2] == 0xFE && buf[3] == 0xFF
+}
+
+func isUTF32LE(buf []byte) bool {
+	return len(buf) >= 4 && buf[0] == 0xFF && buf[1] == 0xFE && buf[2] == 0x00 && buf[3] == 0x00
+}
+
+func isUTF8(buf []byte) bool {
+	return len(buf) >= 3 && buf[0] == 0xEF && buf[1] == 0xBB && buf[2] == 0xBF
+}
+
+func isUTF16BE(buf []byte) bool {
+	return len(buf) >= 2 && buf[0] == 0xFE && buf[1] == 0xFF
+}
+
+func isUTF16LE(buf []byte) bool {
+	return len(buf) >= 2 && buf[0] == 0xFF && buf[1] == 0xFE
+}
+
+// The following variables define the supported encodings.
+// By defining these as package-level variables, we allow easy reference
+// throughout the package and by users of the package. This design also
+// facilitates potential future extension by simply adding new encoding variables.
+
+// Unknown represents an unknown or unrecognized encoding.
+// Having an Unknown encoding allows us to handle cases where
+// the encoding cannot be determined, providing a fallback option.
+var Unknown Encoding = &bomEncoding{
+	encodingType:    "Unknown",
+	sequence:        nil, // Unknown encoding has no BOM sequence
+	hasSequenceFunc: func(data []byte) bool { return false },
+}
+
+// UTF8 represents the UTF-8 encoding.
+var UTF8 Encoding = &bomEncoding{
+	encodingType:    "UTF-8",
+	sequence:        []byte{0xEF, 0xBB, 0xBF}, // UTF-8 BOM sequence
+	hasSequenceFunc: isUTF8,
+}
+
+// UTF16BE represents the UTF-16 Big Endian encoding.
+var UTF16BE Encoding = &bomEncoding{
+	encodingType:    "UTF-16 BE",
+	sequence:        []byte{0xFE, 0xFF}, // UTF-16 BE BOM sequence
+	hasSequenceFunc: isUTF16BE,
+}
+
+// UTF16LE represents the UTF-16 Little Endian encoding.
+var UTF16LE Encoding = &bomEncoding{
+	encodingType:    "UTF-16 LE",
+	sequence:        []byte{0xFF, 0xFE}, // UTF-16 LE BOM sequence
+	hasSequenceFunc: isUTF16LE,
+}
+
+// UTF32BE represents the UTF-32 Big Endian encoding.
+var UTF32BE Encoding = &bomEncoding{
```

Comment:
Minor nit that these could all be value structs instead of pointers since you're not mutating the `bomEncoding` struct after the fact.